### PR TITLE
edit readcourse.data

### PR DESCRIPTION
**Pull request by ak840206**
Is there a way to use *args to initialize course class?

### CODE REVIEW

File: readcourse.data [hunk 1]
```diff
@@ -0,0 +1,148 @@
+import json
+import os
+from pymongo import MongoClient
+
+class Course:
+	def __init__(self, crn):
+		self.crn = crn
+
+	def __init__(self, **kwargs, *args):
+		self.UID = kwargs.['UID']
```
Comment (kis87988) [inline readcourse.data:10]: I don't think this is right.
`kwargs` is a Dict. That means you should put `kwargs['UID']`
In addictional, `kwargs` should be assigned to another one of meaningful variable

File: readcourse.data [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+		self.endTime = kwargs.['endTime']
+		self.cap = kwargs.['cap']
+		self.wl_cap = kwargs.['wl_cap']
+		self.instructor_name = kwargs.['instructor_name']
+		self.startDate = kwargs.['startDate']
+		self.endDate = kwargs.['endDate']
+		self.location = kwargs.['location']
+		self.attribute = kwargs.['attribute']
+    
+//
```
Comment (kis87988) [inline readcourse.data:28]: #? why?

File: readcourse.data [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+		self.wl_act = list()
+
+class Instructor:
+	def __init__(self, firstName, middleName, lastName):
+		self.firstName = firstName
+		self.middleName = middleName
+		self.lastName = lastName
+		self.website = ''
+		self.email = ''
+		self.department = list()
```
Comment (kis87988) [inline readcourse.data:40]: please use `[]` instead `list()`

File: readcourse.data [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+		td.insert_one(temp_dept)
+
+
+
+
+
+
+if __name__ == "__main__":
+	main()
+© 2019 GitHub, Inc.
```
Comment (kis87988) [inline readcourse.data:137]: what is this? please remove it

File: readcourse.data [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+	filename = os.path.join(script_dir, rel_path)
+
+	if filename:
+		with open(filename, 'r') as f:
+			course_raw_data = json.load(f)
+
+	course_list = []
+	department_list = []
+	total_course = 0
+	for department in course_raw_data["2018 Winter De Anza"]["CourseData"]:
```
Comment (kis87988) [inline readcourse.data:75]: should not hard code here

File: readcourse.data [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+		dep_course_num = 0
+		temp_dept = Department(department)
+		for c in course_raw_data["2018 Winter De Anza"]["CourseData"][department]:
```
Comment (kis87988) [inline readcourse.data:78]: same here

File: readcourse.data [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+							 c["Location"], c["Attribute"])
+			course_list.append(temp_course)
+			lack_dept = temp_dept.deptName + " " + temp_course.course_num + " " + temp_course.course_title 
+			if lack_dept not in temp_dept.courses:
+				temp_dept.courses.append(lack_dept)
+			total_course += 1
+		print('department', department, "has", dep_course_num, "sections.")
+		department_list.append(temp_dept)
+
+	print('total course:', total_course)
```
Comment (kis87988) [inline readcourse.data:92]: try to use `logger` if you can

File: readcourse.data [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+	print('loaded course:', len(clist))
+
+	username = input("Enter your username:")
+	password = input("Password:")
+	client = MongoClient("mongodb+srv://" + username +":" + password + "@fhdatimedb-jjsjm.mongodb.net/test?retryWrites=true&w=majority")
```
Comment (kis87988) [inline readcourse.data:97]: this should be factor out a `env` file

**Review by kis87988 — CHANGES_REQUESTED** [review]

**Comment by kis87988** [comment]
Please also check PR #2 comment

File: readcourse.data (hunk 1 above)
Comment (ak840206) [reply readcourse.data:10]: Sorry I don't understand the third line?

File: readcourse.data (hunk 3 above)
Comment (ak840206) [reply readcourse.data:40]: If it change to " self.department = [] ", when using *args or **kwargs, can this become " self.department = args[department]" ?

File: readcourse.data (hunk 5 above)
Comment (ak840206) [reply readcourse.data:75]: Do you mean change ["2018 Winter De Anza"]["CourseData"] to a relative one?

Comment (kis87988) [reply readcourse.data:75]: we should not hard code `2018 Winter De Anza`

File: readcourse.data (hunk 3 above)
Comment (kis87988) [reply readcourse.data:40]: no, just because `[]` are 2 chars, `list()` are 6 chars

File: readcourse.data (hunk 1 above)
Comment (kis87988) [reply readcourse.data:10]: should be `some_variable_name = kargs`

**Pull request closed without merging**